src/Microsoft.AspNet.Authentication/AuthenticationServiceCollectionExtensions.cs

```diff
@@ -12,6 +12,7 @@ public static class AuthenticationServiceCollectionExtensions
     {
         public static IServiceCollection AddAuthentication([NotNull] this IServiceCollection services)
         {
+            services.AddCaching();
```

Comment:
If it's optional then you don't need an extension, caching services would be added when the dev configured caching.